CONTRIBUTING.md

```diff
@@ -144,104 +98,56 @@ into a single one that captures everything you're trying to do -- the history
 will never make for pleasant bedtime reading, so focus instead on making your
 commits useful for tools like `git-blame` and `git-bisect`.
 
-Your contribution must apply cleanly to `master` in order to be considered
-mergeable.  You may want to regularly [rebase your changes][reba] onto `master`
-in order to both clean up any intermediate "development" commits you make and
-to ensure that you're up to date.
+Your contribution must apply cleanly to `develop` in order to be considered
+mergeable.  You may want to regularly [rebase your changes][reba] onto
+`develop` in order to both clean up any intermediate "development" commits you
+make and to ensure that you're up to date.
 
-If you're making a GitHub pull request, it's good practice to make it from a
-topic branch, rather than `master`, on your fork.
+### Pull Requests and Merges
 
-### Pills
+When your work is ready for review, open a pull request, making sure to link to
+the tracking issue in the description, which should be formatted as follows
+(where `<N>` is the number of this work's tracking issue):
 
-Any contribution that touches the kernel (i.e., anything in `pkg/arvo/sys`),
-should be accompanied by an updated [solid pill](#the-kernel-and-pills).  Pills
-are tracked in the repository via [git LFS][git-lfs].
-
-Whenever you make a contribution to the kernel, please create a new solid pill
-via:
-
-```
-sh/update-solid-pill
 ```
+### Description
```

Comment:
This header is superfluous, since you're already writing into the PR description field.

With that, only the "Related" header remains, which seems optional to me, and might also be served by a more modest "See also:" at the start of a new paragraph.

I'd consider putting "resolves #n" at the bottom instead of leading with it, since it's mostly administrative, but this seems more like a personal preference thing.

Reply:
+1 on headers, I think link at the top is good though

Reply:
One nice thing about the header is I think I can set defaults for text fields. Makes it more obvious what's required (and the rendered headers look nice in the field, like in this PR for example).

Reply:
fwiw I personally always bounce super hard on pre-filled PR/issue description fields. Something about them feels, idk, belittling maybe?

Ours would be very minimal, so it's not as bad, but also people should be reading the contributing.md anyway. iirc Github shows a "have your read the contributing.md" for first-time submitters, maybe we can put that in the input field instead? ...But perhaps that's worse. (^: